include/camp/resource/cuda.hpp

```diff
@@ -69,14 +73,16 @@ namespace resources
         return streams[num % 16];
       }
 
+    private:
+      Cuda(cudaStream_t s) : stream(s) {}
     public:
-      Cuda(int group = -1) : stream(get_a_stream(group)) {}
+      Cuda(int group = NEXT_STREAM) : stream(get_a_stream(group)) {}
 
       // Methods
       Platform get_platform() { return Platform::cuda; }
       static Cuda &get_default()
       {
-        static Cuda h;
+        static Cuda h(static_cast<cudaStream_t>(0));
         return h;
```

Comment:
@trws @davidbeckingsale I've been thinking about this and perhaps using a private constructor for setting the stream on the static get_default might be better than allowing users to make multiple default stream cuda objects. What do you think?

Reply:
I'm not sure I quite get what you mean.

Reply:
So instead of having DEFAULT_STREAM and NEXT_STREAM flags we return the Cuda/Hip constructor back to what it was before these changes (removing the change on line 79 and the new stream.hpp file). This will only allow users to get a Cuda resource device with the default stream through the get_default function

Reply:
Oh I see.  That would help make it more like a singleton, it would make it posibly more self-documenting, but harder to write code that will use either a new stream or the default stream based on an argument.

Reply:
Maybe we could have both? :thinking: